### ISSUE TITLE
The first parameter of twisted.internet.threads.blockingCallFromThread should be reactor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3030]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3030|
|Type|enhancement|
|Created|2008-02-12 14:14:53Z|
|Branch|https://github.com/twisted/twisted/tree/blockingcallfromthread-arg-3030|

Global state is bad and stuff.  Parameterization is great and stuff.

Any function which uses a global `reactor` variable should probably be parameterized instead.

This should happen before the next release, since it's not a compatible change.  Fortunately `blockingCallFromThread` hasn't been part of any release yet.

<details><summary>Searchable metadata</summary>

```
trac-id__3030 3030
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__twisted_8_0 twisted-8.0
branch__branches_blockingcallfromthread_arg_3030 branches/blockingcallfromthread-arg-3030
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1202825693000000 1202825693000000
changetime__1203348721000000 1203348721000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [22588]) Branching to 'blockingcallfromthread-arg-3030'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

 * I'm not sure making the reactor import at the top is a great change... whatever
 * please add a test-case-name to wsgi.py

Please merge once done.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [22598]) Merge blockingcallfromthread-arg-3030


Author: exarkun
Reviewer: therve
Fixes [#3030](https://github.com/twisted/twisted/issues/3030)

Change the signature of twisted.internet.threads.blockingCallFromThread to take
the reactor into which it will call as the first parameter.  Update the docs and
the usage in twisted.web2 as well.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
